src/transform/src/cse/map.rs

```diff
@@ -115,13 +115,12 @@ fn memoize(
         _ => {
             // We should not eagerly memoize `if` branches that might not be taken.
             // TODO: Memoize expressions in the intersection of `then` and `els`.
-            if let ScalarExpr::If {
-                cond,
-                then: _,
-                els: _,
-            } = expr
-            {
+            if let ScalarExpr::If { cond, then, els } = expr {
                 memoize(cond, scalars, projection, input_arity);
+                // Conditionally evaluated expressions still need to update their
+                // column references.
+                then.permute(projection);
```

Comment:
I know it’s preexisting at this point but I think `projection` is unnecessarily mutable in the context of this func.